feat: remove query based on addons ping for fenix and rename fenix_addons_by_client_legacy_source_v1 to fenix_addons_by_client_v1 (#9263)
mozilla/bigquery-etl: sql/moz-fx-data-shared-prod/addons_derived/amo_stats_dau_combined_v1/README.md

## Description

> Computes amo stats dau using both the legacy source and the new addons ping data by unioning `moz-fx-data-shared-prod.addons_derived.firefox_desktop_addons_by_client_legacy_source_v1` (legacy telemetry based - includes Firefox version below major 148) and `moz-fx-data-shared-prod.addons_derived.firefox_desktop_addons_by_client_v1` (addons based Glean ping based) for data from major version 148 and above. For Fenix data we use `moz-fx-data-shared-prod.addons_derived.fenix_addons_by_client_v1` (already based on Glean metrics ping - all Firefox versions).

Reviewer: I find the wording slightly confusing and not real need to list the tables here. Someone interested in that detail could check the query.
Also recommending to expand the acronym `amo` here and in the metadata, toi cover for anyone who doesn't know what it means.
```suggestion
Computes AMO (addons.mozilla.org) stats DAU using both the legacy source and the new addons ping data by unioning:
- Firefox Desktop legacy telemetry data for versions before Firefox Desktop major version 148.
- Firefox Desktop Glean addons data including Firefox Desktop versions 148 and above.
- Fenix Glean metrics ping for all Fenix versions.
```

Author: Thanks, implemented most of the suggestion.
